Source/EasyNetQ/Persistent/PersistentConnection.cs

```diff
@@ -62,24 +65,32 @@ public void Dispose()
     {
         if (disposed) return;
 
-        disposed = true;
-
         DisposeConnection();
+        disposed = true;
     }
 
     private IAutorecoveringConnection InitializeConnection()
     {
         var connection = initializedConnection;
-        if (connection != null) return connection;
+        if (connection is not null) return connection;
 
-        lock (mutex)
+        try
         {
-            connection = initializedConnection;
-            if (connection != null) return connection;
+            lock (mutex)
+            {
+                connection = initializedConnection;
+                if (connection is not null) return connection;
 
-            connection = initializedConnection = CreateConnection();
+                connection = initializedConnection = CreateConnection();
+            }
+        }
+        catch (Exception exception)
+        {
+            status = status.ToDisconnected(exception.Message);
```

Comment:
there are no tests for this behaviour. is it ok?

Reply:
It is not true. Please comment this line and you will see broken tests.